cobalt/browser/cobalt_content_browser_client.cc

```diff
@@ -80,6 +88,56 @@ constexpr base::FilePath::CharType kTransportSecurityPersisterFilename[] =
 constexpr base::FilePath::CharType kTrustTokenFilename[] =
     FILE_PATH_LITERAL("Trust Tokens");
 
+// An enum for the UMA histogram to track the state of the variations config.
+// These values are persisted to logs. Entries should not be renumbered and
+// numeric values should never be reused.
+enum class VariationsConfigState {
+  kValid = 0,
+  kExpired = 1,
+  kMissingTimestamp = 2,
+  kMaxValue = kMissingTimestamp,
+};
+
+// Checks if the experiment config has expired by reading the last fetch time.
+bool HasConfigExpired() {
+  PrefService* experiment_prefs =
+      GlobalFeatures::GetInstance()->experiment_config();
+
+  // If the pref is missing, we cannot determine its age. For backward
+  // compatibility and safety on first run, we treat it as valid
+  if (!experiment_prefs->HasPrefPath(
+          variations::prefs::kVariationsLastFetchTime)) {
+    UMA_HISTOGRAM_ENUMERATION("Cobalt.Finch.ConfigState",
+                              VariationsConfigState::kMissingTimestamp);
+    return false;
+  }
+
+  base::Time fetch_time =
+      experiment_prefs->GetTime(variations::prefs::kVariationsLastFetchTime);
+  base::TimeDelta config_age = base::Time::Now() - fetch_time;
+
+  UMA_HISTOGRAM_CUSTOM_COUNTS("Cobalt.Finch.ConfigAgeInDays",
+                              config_age.InDays(), 1, 90, 50);
```

Comment:
super-nit: Should we define the custom count MACRO to something like "UMA_HISTOGRAM_DAYS_ELAPSED"? Mostly to add notations around the values range? No need if this is widely known on the cobalt side!